src/consensus/encode.rs

```diff
@@ -306,6 +308,24 @@ pub trait Encodable {
     ///
     /// The only errors returned are errors propagated from the writer.
     fn consensus_encode<W: io::Write + ?Sized>(&self, writer: &mut W) -> Result<usize, io::Error>;
+
+    /// Returns the bytes needed when `self` is serialized
+    ///
+    /// The computation could be costly for big objects, see [`Encodable::serialized_len_early_stop`] to limit
+    /// the cost at the expense of precision for big objects
+    fn serialized_len(&self) -> usize {
+        self.consensus_encode(&mut EmptyWrite).unwrap()
```

Comment:
Nice trick, but damn, I'd love to force Rust into doing const propagation. Impossible until `const` in traits and `&mut` in `const` context are implemented though.